godot-core/src/builtin/vector_macros.rs

```diff
@@ -0,0 +1,275 @@
+    (
+        // Name of the vector type.
+        $Vector:ty,
+        // Type of each individual component, for example `i32`.
+        $Scalar:ty,
+        // Names of the components, with parentheses, for example `(x, y)`.
+        ($($components:ident),*),
+        // Name of the operator trait, for example `AddAssign`.
+        $Operator:ident,
+        // Name of the function on the operator trait, for example `add_assign`.
+        $func:ident$(,)?
+    ) => {
+        impl std::ops::$Operator for $Vector {
+            fn $func(&mut self, rhs: $Vector) {
+                $(
+                    self.$components.$func(rhs.$components);
+                )*
+            }
+        }
+    }
+}
+
+/// Implements a single arithmetic assignment operator for a vector type, with a scalar on the
+/// right-hand side.
+macro_rules! impl_vector_scalar_assign_operator {
+    (
+        // Name of the vector type.
+        $Vector:ty,
+        // Type of each individual component, for example `i32`.
+        $Scalar:ty,
+        // Names of the components, with parentheses, for example `(x, y)`.
+        ($($components:ident),*),
+        // Name of the operator trait, for example `AddAssign`.
+        $Operator:ident,
+        // Name of the function on the operator trait, for example `add_assign`.
+        $func:ident$(,)?
+    ) => {
+        impl std::ops::$Operator<$Scalar> for $Vector {
+            fn $func(&mut self, rhs: $Scalar) {
+                $(
+                    self.$components.$func(rhs);
+                )*
+            }
+        }
+    }
+}
+
+/// Implements all common arithmetic operators on a built-in vector type.
+macro_rules! impl_vector_operators {
+    (
+        // Name of the vector type to be implemented, for example `Vector2`.
+        $Vector:ty,
+        // Type of each individual component, for example `f32`.
+        $Scalar:ty,
+        // Names of the components, with parentheses, for example `(x, y)`.
+        ($($components:ident),*)$(,)?
+    ) => {
+        impl_vector_unary_operator!($Vector, $Scalar, ($($components),*), Neg, neg);
+        impl_vector_vector_binary_operator!($Vector, $Scalar, ($($components),*), Add, add);
+        impl_vector_vector_binary_operator!($Vector, $Scalar, ($($components),*), Sub, sub);
+        impl_vector_vector_binary_operator!($Vector, $Scalar, ($($components),*), Mul, mul);
+        impl_vector_scalar_binary_operator!($Vector, $Scalar, ($($components),*), Mul, mul);
+        impl_scalar_vector_binary_operator!($Vector, $Scalar, ($($components),*), Mul, mul);
+        impl_vector_vector_binary_operator!($Vector, $Scalar, ($($components),*), Div, div);
+        impl_vector_scalar_binary_operator!($Vector, $Scalar, ($($components),*), Div, div);
+        impl_vector_vector_binary_operator!($Vector, $Scalar, ($($components),*), Rem, rem);
+        impl_vector_scalar_binary_operator!($Vector, $Scalar, ($($components),*), Rem, rem);
```

Comment:
As mentioned, I would remove the `%` ones, as Godot doesn't have them and there is no intuitive understanding.

Reply:
Done.